tapfreighter/wallet.go

```diff
@@ -1654,61 +1654,25 @@ func addAnchorPsbtInputs(btcPkt *psbt.Packet, vPkt *tappsbt.VPacket,
 	}
 
 	// Now that we've added an extra input, we'll want to re-calculate the
-	// total weight of the transaction, so we can ensure we're paying
-	// enough in fees.
-	var (
-		weightEstimator     input.TxWeightEstimator
-		inputAmt, outputAmt int64
-	)
-	for _, pIn := range btcPkt.Inputs {
-		inputAmt += pIn.WitnessUtxo.Value
-
-		inputPkScript := pIn.WitnessUtxo.PkScript
-		switch {
-		case txscript.IsPayToWitnessPubKeyHash(inputPkScript):
-			weightEstimator.AddP2WKHInput()
-
-		case txscript.IsPayToScriptHash(inputPkScript):
-			weightEstimator.AddNestedP2WKHInput()
-
-		case txscript.IsPayToTaproot(inputPkScript):
-			weightEstimator.AddTaprootKeySpendInput(
-				txscript.SigHashDefault,
-			)
-		default:
-			return fmt.Errorf("unknown pkScript: %x",
-				inputPkScript)
+	// total vsize of the transaction and the necessary fee at the desired
+	// fee rate.
+	inputScripts := make([][]byte, 0, len(btcPkt.Inputs))
+	for inputIdx, input := range btcPkt.Inputs {
+		if input.WitnessUtxo.PkScript == nil {
```

Comment:
Should check `input.WitnessUtxo` being `nil` first, as that's not mandatory.

Reply:
Agree, but is there a 'correct' case where `input.WitnessUtxo` is non-nil? IIUC any change outputs added should have this field populated, and earlier in this func we populate it for anchor inputs.

Reply:
In `lnd` we always set `WitnessUtxo`, but it's not strictly mandatory according to the BIP. But since most hardware wallets also require it, we can assume that any software creating a PSBT would set it. So we can error out if it's `nil`.

Reply:
Sounds good, I imagine this would also come up if the funding input is not-segwit like P2SH?

(Which wouldn't come from lnd 0.17+, but may be a consideration once we support external inputs).

Reply:
`lnd` doesn't create any non-witness addresses, so it should never have non-witness UTXOs. Even nested-p2wkh (wrapped in p2wh) is segwit and therefore needs to have the witness UTXO field set.

Reply:
Added a TODO + note on #118 where I expect this to be most relevant.